SYCL/ESIMD/lsc/lsc_load2d_compare.cpp

```diff
@@ -0,0 +1,103 @@
+//==------------ load2d_compare.cpp - DPC++ ESIMD on-device test ----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu-intel-pvc || esimd_emulator
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// The tests makes sure old and new load_2d API produce identical results.
+
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace __ESIMD_NS;
+using namespace __ESIMD_ENS;
+
+int main() {
+  constexpr uint32_t BlockWidth = 16;
+  constexpr uint32_t BlockHeight = 4;
+  constexpr uint32_t NumBlocks = 1;
+  constexpr uint32_t SurfaceHeight = 16;
+  constexpr uint32_t SurfaceWidth = 16;
+  constexpr uint32_t SurfacePitch = 16;
+  constexpr uint32_t x = 0;
+  constexpr uint32_t y = 0;
+
+  constexpr uint32_t Size = SurfacePitch * SurfaceHeight * NumBlocks;
+
+  queue q;
+  auto dev = q.get_device();
+  std::cout << "Running on " << dev.get_info<info::device::name>() << "\n";
+  auto ctxt = q.get_context();
+  auto *A =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+  auto *B =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+  auto *C =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+  auto *C1 =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
```

Comment:
Minor comment: this could be written simpler:
```suggestion
  auto *A = malloc_shared<float>(Size, q);
  auto *B = malloc_shared<float>(Size, q);
  auto *C = malloc_shared<float>(Size, q);
  auto *C1 = malloc_shared<float>(Size, q);
```

Reply:
Fixed